src/extension/background-script/actions/nostr/signEventOrPrompt.ts

```diff
@@ -17,15 +19,44 @@ const signEventOrPrompt = async (message: MessageSignEvent) => {
     };
   }
 
+  const allowance = await db.allowances.get({
+    host: message.origin.host,
+  });
+
+  if (!allowance?.id) {
+    return { error: "Could not find an allowance for this host" };
+  }
+
+  const findPermission = await db.permissions.get({
+    host: message.origin.host,
+    method: PermissionMethod["NOSTR_SIGNMESSAGE"],
+  });
+
+  const hasPermission = !!findPermission?.enabled;
+
   try {
-    const response = await utils.openPrompt<{
-      confirm: boolean;
-    }>({
-      ...message,
-      action: "public/nostr/confirmSignMessage",
-    });
-    if (!response.data.confirm) {
-      throw new Error("User rejected");
+    if (!hasPermission) {
+      const promptResponse = await utils.openPrompt<{
+        confirm: boolean;
+        rememberPermission: boolean;
```

Comment:
nit: in ln/request we call it `enabled` (which is probably also not a good name)
https://github.com/getAlby/lightning-browser-extension/blob/master/src/extension/background-script/actions/ln/request.ts#L67-L68